haystack/document_store/elasticsearch.py

```diff
@@ -225,14 +225,46 @@ def get_document_by_id(self, id: str, index=None) -> Optional[Document]:
         else:
             return None
 
-    def get_documents_by_id(self, ids: List[str], index=None) -> List[Document]:
+    def get_documents_by_id(self, ids: List[str], index: Optional[str] = None) -> List[Document]:
         """Fetch documents by specifying a list of text id strings"""
         index = index or self.index
         query = {"query": {"ids": {"values": ids}}}
         result = self.client.search(index=index, body=query)["hits"]["hits"]
         documents = [self._convert_es_hit_to_document(hit, return_embedding=self.return_embedding) for hit in result]
         return documents
 
+    def get_metadata_values_by_key(
+        self,
+        key: str,
+        query: Optional[str] = None,
+        filters: Optional[Dict[str, List[str]]] = None,
+        index: Optional[str] = None,
+    ) -> List[dict]:
+        """
+        Get values associated with a metadata key. The output is in the format:
+            [{"value": "my-value-1", "count": 23}, {"value": "my-value-2", "count": 12}, ... ]
+        """
```

Comment:
Can you please add the params to the docstring? We should slowly increase our coverage here so that the docs look nicer on the website.